Essentials/src/com/earth2me/essentials/Essentials.java

```diff
@@ -357,12 +373,29 @@ private void registerListeners(PluginManager pm) {
 
     @Override
     public void onDisable() {
+        boolean stopping = false;
+        if (isStopping != null) {
+            try {
+                stopping = (boolean) isStopping.invoke();
+            } catch (Throwable throwable) {
+                throwable.printStackTrace();
+            }
+        }
         for (User user : getOnlineUsers()) {
             if (user.isVanished()) {
                 user.setVanished(false);
                 user.sendMessage(tl("unvanishedReload"));
             }
-            user.stopTransaction();
+            if (stopping) {
+                user.setLogoutLocation();
+                if (!user.isHidden()) {
+                    user.setLastLogout(System.currentTimeMillis());
+                }
+                user.stopTransaction();
+                user.cleanup();
```

Comment:
@md678685 shouldn't this be called anyways (regardless of shutdown)? Could see instances where this isn't called on Disable due to a /reload or plugman bullshit where we lose data.

Reply:
`stopTransaction` and `cleanup` both run a save task on another thread, but `cleanup` waits for that task to complete. Running cleanup during a reload means the server will hang up while saving data for everyone, which isn't necessary. I'm pretty sure loading the user data would wait for the save task regardless, if we actually *are* reloading.